include/internal/common.h

```diff
@@ -93,87 +93,170 @@ __owur static inline int ossl_assert_int(int expr, const char *exprstr,
-                         *((c)++)=(unsigned char)(((l)    )&0xff))
-
-# define l2n8(l,c)       (*((c)++)=(unsigned char)(((l)>>56)&0xff), \
-                         *((c)++)=(unsigned char)(((l)>>48)&0xff), \
-                         *((c)++)=(unsigned char)(((l)>>40)&0xff), \
-                         *((c)++)=(unsigned char)(((l)>>32)&0xff), \
-                         *((c)++)=(unsigned char)(((l)>>24)&0xff), \
-                         *((c)++)=(unsigned char)(((l)>>16)&0xff), \
-                         *((c)++)=(unsigned char)(((l)>> 8)&0xff), \
-                         *((c)++)=(unsigned char)(((l)    )&0xff))
-
-/* NOTE - c is not incremented as per l2c */
-# define l2cn(l1,l2,c,n) { \
-                        c+=n; \
-                        switch (n) { \
-                        case 8: *(--(c))=(unsigned char)(((l2)>>24)&0xff); \
-                        case 7: *(--(c))=(unsigned char)(((l2)>>16)&0xff); \
-                        case 6: *(--(c))=(unsigned char)(((l2)>> 8)&0xff); \
-                        case 5: *(--(c))=(unsigned char)(((l2)    )&0xff); \
-                        case 4: *(--(c))=(unsigned char)(((l1)>>24)&0xff); \
-                        case 3: *(--(c))=(unsigned char)(((l1)>>16)&0xff); \
-                        case 2: *(--(c))=(unsigned char)(((l1)>> 8)&0xff); \
-                        case 1: *(--(c))=(unsigned char)(((l1)    )&0xff); \
-                                } \
-                        }
-
-# define n2s(c,s)        ((s=(((unsigned int)((c)[0]))<< 8)| \
-                             (((unsigned int)((c)[1]))    )),(c)+=2)
-# define s2n(s,c)        (((c)[0]=(unsigned char)(((s)>> 8)&0xff), \
-                           (c)[1]=(unsigned char)(((s)    )&0xff)),(c)+=2)
-
-# define n2l3(c,l)       ((l =(((unsigned long)((c)[0]))<<16)| \
-                              (((unsigned long)((c)[1]))<< 8)| \
-                              (((unsigned long)((c)[2]))    )),(c)+=3)
-
-# define l2n3(l,c)       (((c)[0]=(unsigned char)(((l)>>16)&0xff), \
-                           (c)[1]=(unsigned char)(((l)>> 8)&0xff), \
-                           (c)[2]=(unsigned char)(((l)    )&0xff)),(c)+=3)
+/*
+ * Loop rolling/unrolling, use by some cipher implementations.
+ * Note that all of these evaluate their parameters multiple
+ * times. We could remove the parens around some of them,
+ * like "(c)" etc., but don't. Many of the macros use the comma
+ * to get a "sequence point" to ensure that the right bytes
+ * are shifted with the right amounts.
+ */
+
+# define n2s(c, s) \
+    ((s = (((unsigned int)((c)[0])) << 8) | \
+          (((unsigned int)((c)[1]))     )), \
+          (c) += 2)
+
+# define s2n(s, c) \
+    (((c)[0] = (unsigned char)(((s) >> 8) & 0xff), \
+      (c)[1] = (unsigned char)(((s)     ) & 0xff)), \
+      (c) += 2)
+
+# define n2l3(c, l) \
+    ((l = (((unsigned long)((c)[0])) <<16) | \
+          (((unsigned long)((c)[1])) << 8) | \
+          (((unsigned long)((c)[2]))    )), \
```

Comment:
The first two lines should have an extra space after `<<` and the third line then needs an extra space for alignment.

Reply:
fixed